packages/app-desktop/gui/MainScreen/commands/perNotebookSortOrder.ts

```diff
@@ -0,0 +1,212 @@
+import Setting from '@joplin/lib/models/Setting';
+import { State } from '@joplin/lib/reducer';
+import app from '../../../app';
+import eventManager from '@joplin/lib/eventManager';
+import {
+	notesSortOrderFieldArray, NOTES_SORT_ORDER_SWITCH, SETTING_FIELD, SETTING_REVERSE, SETTING_PER_FIELD_REVERSAL_ENABLED,
+} from './notesSortOrderSwitch';
+import { _ } from '@joplin/lib/locale';
+
+
+export const PER_NOTEBOOK_SORT_ORDER = 'perNotebookSortOrder';
+export const SETTING_PER_NOTEBOOK_SORT_ORDER_ENABLED = 'notes.perNotebookSortOrderEnabled';
+
+let previousNotebookId: string = null;
+let ownSortOrders: { [key: string]: any } = null;
+const sharedSortOrder: { [key: string]: any } = {
+	field: 'user_updated_time',
+	reverse: true,
+	user_updated_time: true,
+	user_created_time: true,
+	title: false,
+	order: false,
+};
+
+export function selectedNotebookHasOwnSortOrder() {
+	return hasOwnSortOrder(getSelectedNotebookId());
+}
+
+export function hasOwnSortOrder(notebookId: string) {
+	return notebookId && ownSortOrders && ownSortOrders.hasOwnProperty(notebookId + SUFFIX_FIELD);
+}
+
+export const declaration: CommandDeclaration = {
+	name: PER_NOTEBOOK_SORT_ORDER,
+	label: () => _('Toggle own sort order'),
+};
+
+export const runtime = (): CommandRuntime => {
+	loadOwnSortOrders();
+	loadSharedSortOrder();
+	eventManager.appStateOn('notesParentType', onNotebookSelectionMayChange);
+	eventManager.appStateOn('selectedFolderId', onNotebookSelectionMayChange);
+
+	return {
+		enabledCondition: 'oneFolderSelected',
+
+		execute: async (context: CommandContext, notebookId?: string, own?: boolean) => {
+			let targetId = notebookId;
+			const selectedId = getSelectedNotebookId(context.state);
+			if (!targetId) {
+				targetId = selectedId; // default: selected notebook
+				if (!targetId) return;
+			}
+			const targetOwn = hasOwnSortOrder(targetId);
+			let newOwn;
+			if (typeof own == 'undefined') {
+				newOwn = !targetOwn; // default: toggling
+			} else {
+				newOwn = !!own;
+				if (newOwn === targetOwn) return;
+			}
+			if (newOwn) {
+				let field: string, reverse: boolean;
+				if (!hasOwnSortOrder(selectedId)) {
+					field = getCurrentField();
+					reverse = getCurrentReverse();
+				} else {
+					field = sharedSortOrder.field;
+					if (perFieldReversalEnabled()) {
+						reverse = sharedSortOrder[field];
+					} else {
+						reverse = sharedSortOrder.reverse;
+					}
+				}
+				setOwnSortOrder(targetId, field, reverse);
+			} else {
+				deleteOwnSortOrder(targetId);
+			}
+		},
+	};
+};
+
+function onNotebookSelectionMayChange() {
+	const selectedId = getSelectedNotebookId();
+	if (previousNotebookId === null) previousNotebookId = selectedId;
+	if (previousNotebookId === selectedId) return;
+	const field = getCurrentField();
+	const reverse = getCurrentReverse();
+	const previousNotebookHasOwnSortOrder = hasOwnSortOrder(previousNotebookId);
```

Comment:
previousFolderHasOwnSortOrder

There are a few others so please go through the code and rename to "folder" every time. It's something I'm trying to keep consistent even though there's an inconsistency between UI (notebook) and backend (folder).

Reply:
finished. c723ebe